### PR TITLE
select2 upgrade and skin editor tweaks

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: card/mod/bootstrap/set/type_plus_right/customized_skin/colors.rb [hunk 1]
```diff
@@ -97,5 +100,5 @@ def replace_values group, prefix=""
 end
 
 def variable_values_from_params group
-  Env.params[group].slice(*VARIABLE_NAMES[group])
+  Env.params.dig(:group)&.slice(*VARIABLE_NAMES[group]) || {}
```
Comment (ethn) [inline card/mod/bootstrap/set/type_plus_right/customized_skin/colors.rb]: Ooh, I like "slice"

File: decko/decko.gemspec [hunk 2]
```diff
@@ -34,7 +34,6 @@ Gem::Specification.new do |s|
   s.required_ruby_version = ">= 2.3"
 
   [
-    ["rails", "~> 5.1"],
```
Comment (ethn) [inline decko/decko.gemspec]: because it's in card?

File: card/mod/bootstrap/set/type_plus_right/customized_skin/colors.rb [hunk 3]
```diff
@@ -13,30 +13,32 @@
 # TODO: deal with that
 
 def variable_value name
```
Comment (ethn) [inline card/mod/bootstrap/set/type_plus_right/customized_skin/colors.rb]: maybe a comment that "name" can SCSS (and is treated as such if it starts with a "$")?

File: card/mod/pointer/set/abstract/code_pointer.rb [hunk 4]
```diff
@@ -3,7 +3,8 @@
 abstract_basket :item_codenames
 
 def content
-  item_codenames.map do |codename|
-    Card[codename] && Card[codename].name
-  end.compact.to_pointer_content
+  @cached_content ||=
```
Comment (ethn) [inline card/mod/pointer/set/abstract/code_pointer.rb]: what clears this?

File: decko/decko.gemspec (hunk 2 above)
Comment (xithan) [reply decko/decko.gemspec]: y, I moved it to card and changed it to < 5.2

File: card/mod/pointer/set/abstract/code_pointer.rb (hunk 4 above)
Comment (xithan) [reply card/mod/pointer/set/abstract/code_pointer.rb]: `item_codenames` can only be changed with code but maybe you can break it if you change the names of the item cards

**Pull request merged**